### ISSUE TITLE
Deep River graphics

### ISSUE THREAD

**Issue by Tadrix** [endless-sky/endless-sky#2216]
Would it be possible to dynamically change an image for Deep River depending on how full with cargo it is?
e.g.:
0% — deep river 0.png
1%–20% — deep river 1.png
21%–40% — deep river 2.png
41%–60% — deep river 3.png
61%–80% — deep river 4.png
81%–100% — deep river.png
Meaning, player comes into spaceport with full Deep River, sells all of it's cargo, then buys another commodity (but due to insufficient funds while buying luxury goods fills only some part of its hold) and then departs, his Deep River would look like deep river 3.png, for example, until more cargo would be picked up. Or is it a silly idea?

**Comment by KiLEdEnNis** [comment]
in this state it's not possible since the stats actually change: max cargo & drag
if that can be worked around it seems like a good idea

**Comment by jafdy** [comment]
I would be a nice touch but it is a lot of effort for an entirely visual change.

**Comment by KiLEdEnNis** [comment]
although it could enable more ships with and without cargo pods, like the Arach Spindle/Hulk (& my 3 Freighters in my plugin)
but if it'll even happen, it won't happen anytime soon

**Comment by Amazinite** [comment]
Where are you getting and dropping off the crates? They're Wanderer made storage crates at first, but if you're trading in human space then where do you get them? And what if you dump cargo in space, do the crates just disappear or fly off?

**Comment by Bladewood** [comment]
@Amazinite My thoughts exactly. And the same thing would apply to ships like the Hulk/Spindle.

**Comment by Tadrix** [comment]
Funny how dumped cargo (at least commodities) looks identical no matter where player bought it, now that you said that…

I'm assuming that we really know how Wanderer cargo pods really work. They could all be there but just used ones are unfolded or inflated, for example. And, if player sells that cargo, (s)he'd just fold 'em and store somewhere in the ship, or deflate them (in which case they store themselves in ship's axis beam).

Of course, if this would add much more work it wouldn't be feasible, but that seemed like a good idea.

**Comment by MessyMix** [comment]
@Tadrix: You're entirely correct. It does (did?) seem like a good idea, but
perhaps a) it's not consistent with lore, as @Amazinite pointed out, and b)
it's a lot of work.

**Comment by Tadrix** [comment]
Yeah, I meant "don't really know how these pods work". Or is there lore about Deep River's operation?

**Comment by KiLEdEnNis** [comment]
@Tadrix in data, read the ship descriptions etc.

**Comment by Tadrix** [comment]
@KiLEdEnNis Nah, I wanted to know about ship prefixes and Kor Sestor abbreviations… For example, Republic Navy ships usually have prefix R.N.S., but Oathkeepers differ in a non-consistent way: admiral Danforth's cruiser is always named "N.S. Peacemaker", and other cruiser (in reconciliation ending) is always named "R.S. Hand of Justice", so I'd like to somehow clarify if this is supposed to be this way and why they differ.

**Comment by Amazinite** [comment]
Republic Navy Ship (Part of the Republic and of the Navy)
Navy Ship (Only in the Navy,m)
Republic Ship (Only in the Republic, but not part of the Navy)

**Comment by Amazinite** [comment]
Republic Navy Ship (Part of the Republic and of the Navy)
Navy Ship (In the Navy, but not under the command of the Republic)
Republic Ship (In the Republic, but not part of the Navy)

**Comment by Tadrix** [comment]
Sorry… Oathkeeper cruiser not part of the Navy? How that's possible?

**Comment by KiLEdEnNis** [comment]
the Oathkeepers are a different organization, they're a special unit under direct Navy command (by Admiral Danforth), if the Republic had control over them they could deploy the Oathkeepers against the Free Worlds

**Comment by Tadrix** [comment]
Ok, then we can agree that all Oathkeepers are Navy (it's in the name, Navy (Oathkeepers)). Then what's with the ship "R.S. Hand of Justice"? It is an Oathkeeper cruiser. If you are right, then all Oathkeeper vessels should have prefix "N.S." Prefix "R.S." strikes me as something an organization that answers only to the Parliament directly would have.

Besides, I suspect this situation is more complicated. All of the Navy is under Parliament control. It's interesting what would happen if Republic ordered all of them to be deployed to the South (if need to follow orders, or need to abide by oath taken would precede — that's some samurai dilemmas here), but that would never happen since Republic cares about keeping Paradise planets safe from pirates, so they most likely don't care about Oathkeepers as long as sufficient part of the Navy gets deployed where they want and the job gets done.

**Comment by jafdy** [comment]
I think this has gone off-topic by a ~~bit~~ lot. If you want to continue discussing it then can you move over to Steam/Google groups/Discord.

**Comment by Tadrix** [comment]
That's what I asked for previously:
> BTW, where is the best place to ask questions about lore? Somewhere in Google Groups?

Where exactly is the best place to ask questions about lore?

**Comment by KiLEdEnNis** [comment]
try google groups because github is for features etc.

**Comment by Amazinite** [comment]
>They could all be there but just used ones are unfolded or inflated, for example. 

*Picks up cargo while in space*
*Suddenly, at the speed of light, my cargo pods expand*

**Comment by Tadrix** [comment]
@KiLEdEnNis Ok, google groups then. Thanks.

@Amazinite Yes, and my ship instantly sucks such cargo in without any cranes, tractor beams, etc. right through walls, engine nozzles and torpedo bays. :)

Collapsible cargo containers are becoming a reality in our age already. There's no reason to believe some advanced civilization (such as Wanderers) couldn't use sturdier version which is even faster set up in their starships.

**Comment by Amazinite** [comment]
That point I was making is that you would need to make animations.

**Comment by Tadrix** [comment]
Perhaps I could try my hand at this (although I have absolutely no experience at animating, modeling or image processing). The process would be fraction of a second long, so it wouldn't have to be especially detailed either.

**Comment by endless-sky** [comment]
This really doesn't seem worthwhile, and as others have mentioned, it would raise the question of where you're storing the cargo pods when they aren't in use.
